deploy-board/deploy_board/static/js/components/capacitycomponents.js

```diff
@@ -214,30 +257,95 @@ Vue.component("static-capacity-config", {
-                </div>\
-            </div>\
-    </div>',
-    props: ['minsize', 'maxsize'],
-    methods: {
-        updateMinSize: function (value) {
-            this.$emit('minchange', value)
+    template: `<div>
+    <div class="form-group">
+        <label for="capacity" data-toggle="tooltip" class="deployToolTip control-label" :class="labelBootstrapClass" :title="labelText">
+            {{ labelTitle }}
+        </label>
+        <div :class="inputBootstrapClass">
+            <div class="input-group">
+                <span class="input-group-addon">Min Size</span>
+                <input name="minSize" class="form-control" type="number" min="0" required
+                    :value="minSize" @input="onMinSizeChange($event.target.value)" @keydown.enter.prevent="" >
+            </div>
+        </div>
+        <div :class="inputBootstrapClass">
+            <div class="input-group">
+                <span class="input-group-addon">Max Size</span>
+                <input name="maxSize" class="form-control" type="number" min="0" required
+                    :value="maxSize" @input="onMaxSizeChange($event.target.value)" @keydown.enter.prevent="">
+            </div>
+        </div>
+    </div>
+    <form-warning v-show="showSizeWarning" :alert-text="sizeWarning"></form-warning>
+    <form-error v-show="showSizeError" :alert-text="sizeError"></form-error>
+    </div>`,
+    props: {
+        labelBootstrapClass: {
+            type: String,
+            default: 'col-xs-4',
+        },
+        inputBootstrapClass: {
+            type: String,
+            default: 'col-xs-2',
+        },
+        labelText: {
+            type: String,
+            default: 'Number of hosts for this service',
         },
-        updateMaxSize: function (value) {
-            this.$emit('maxchange', value)
+        labelTitle: {
+            type: String,
+            default: 'Capacity'
+        },
+        originalMinSize: Number,
+        originalMaxSize: Number,
+        remainingCapacity: Number,
+        placements: Object,
+    },
+    data: function() {
+        return {
+            minSize: this.originalMinSize,
+            maxSize: this.originalMaxSize,
+            showSizeError: false,
+            showSizeWarning: false,
+            sizeError: '',
+            sizeWarning: '',
         }
+    },
+    methods: {
+        onMinSizeChange: function (value) {
+            this.minSize = Number(value);
+            if (this.maxSize < this.minSize) {
+                this.maxSize = this.minSize;
+            }
+            this.validateSize();
+            this.$emit('min-change', this.minSize);
+        },
+        onMaxSizeChange: function (value) {
+            this.maxSize = Number(value);
+            if (this.maxSize < this.minSize) {
+                this.minSize = this.maxSize;
+            }
+            this.validateSize();
+            this.$emit('max-change', this.maxSize);
+        },
+        validateSize: function() {
```

Comment:
Can we simplify this logic? If we are not preventing users from making this modification, we can just warn for these cases.

Reply:
I'm not sure if I follow. Here it is trying to warn users inside `validateSize`.

Reply:
I meant we can have generic warning message when capacity is changed such that ASG could have imbalance to simplify logic that's in generateCapacityAlertMsg. 

Also, ASG could run into capacity issue without this config change. So, I think checking this condition on capacity page load would be great.

Reply:
also, it seems it would catch total capacity issues in ASG and not the AZ imbalance ? Is that the intention?

Reply:
That is the intention. Regarding AZ imbalance, I think we can check `if (maxSize - current size) / numPlacements >= any remaining capacity`. However this will inevitably result in a even complicated logic.

Reply:
Done